File: frontend/src/types/schedule.ts
export interface Schedule {
  id: string
  workflow_id: string
  name: string
  cron_expression: string
  is_active: boolean
  last_run_at?: string
  next_run_at?: string
  created_at: string
  updated_at: string
}

export interface ScheduleCreate {
  workflow_id: string
  name: string
  cron_expression: string
  is_active?: boolean
}

export interface ScheduleUpdate {
  name?: string
  cron_expression?: string
  is_active?: boolean
}

File: backend/app/schemas/schedule.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class ScheduleCreate(BaseModel):
    workflow_id: str
    name: str = Field(..., min_length=1, max_length=255)
    cron_expression: str = Field(..., description="Cron ifadesi (örn: '0 2 * * *')")
    is_active: bool = True


class ScheduleUpdate(BaseModel):
    name: Optional[str] = None
    cron_expression: Optional[str] = None
    is_active: Optional[bool] = None


class ScheduleResponse(BaseModel):
    id: str
    workflow_id: str
    name: str
    cron_expression: str
    is_active: bool
    last_run_at: Optional[datetime] = None
    next_run_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
